testng-core-api/src/main/java/org/testng/ITestResult.java

```diff
@@ -114,6 +115,29 @@
    */
   String id();
 
+  /**
+   * @return - <code>true</code> if the current test result is either {@link ITestResult#STARTED} or
+   *     {@link ITestResult#CREATED}
+   */
+  default boolean isStatusCreatedOrStarted() {
+    return getStatus() == STARTED || getStatus() == CREATED;
+  }
+
+  /**
+   * @return - A <code>|</code> symbol separated values which includes all user facing statuses
+   *     viz.,
+   *     <ul>
+   *       <li>{@link ITestResult#SUCCESS}
+   *       <li>{@link ITestResult#SUCCESS_PERCENTAGE_FAILURE}
+   *       <li>{@link ITestResult#FAILURE}
+   *       <li>{@link ITestResult#SKIP}
+   *     </ul>
+   */
+  static String validUserFacingStatuses() {
```

Comment:
Could be `finalStatuses()`

Why not introducing en enum?

Reply:
Fixed. I chose not to introduce an enum because the underlying TestNG statuses are still backed by integers.